task-driver/integration/tests/settle_match.rs

```diff
@@ -82,41 +81,25 @@ async fn setup_wallet_with_order_balance(
     balance: Balance,
     test_args: IntegrationTestArgs,
 ) -> Result<(Wallet, Scalar, Scalar)> {
-    let client = &test_args.arbitrum_client;
-    let account_addr = biguint_from_address(client.wallet_address());
-    let (mut wallet, blinder_seed, share_seed) = new_wallet_in_darkpool(client).await?;
-
-    // Deposit the balance into the wallet
-    let old_wallet = wallet.clone();
-    wallet.balances.insert(balance.mint.clone(), balance.clone());
-    wallet.reblind_wallet();
+    let mut rng = thread_rng();
 
-    wallet = execute_wallet_update(
-        old_wallet,
-        wallet.clone(),
-        Some(ExternalTransfer {
-            mint: balance.mint,
-            amount: balance.amount.into(),
-            account_addr,
-            direction: ExternalTransferDirection::Deposit,
-        }),
-        test_args.clone(),
-    )
-    .await?;
+    let blinder_seed = Scalar::random(&mut rng);
+    let share_seed = Scalar::random(&mut rng);
+    let mut wallet = mock_empty_wallet();
 
-    // Add the order to the wallet
-    let old_wallet = wallet.clone();
+    // Add the balance and order into the wallet
+    wallet.balances.insert(balance.mint.clone(), balance.clone());
     wallet.orders.insert(Uuid::new_v4(), order);
-    wallet.reblind_wallet();
-
-    wallet = execute_wallet_update(
-        old_wallet,
-        wallet.clone(),
-        None, // transfer
-        test_args.clone(),
-    )
-    .await?;
+    setup_initial_wallet(blinder_seed, share_seed, &mut wallet, &test_args).await?;
```

Comment:
just so we're on the same page: this will only work while verification is disabled (since the proof of `VALID WALLET CREATE` should not verify). this means once we re-enable verification we'll have to reimplement this method to reblind & update the wallet on-chain as it used to do

Reply:
It'd be preferable to expose a contract mock method that allowed direct allocation. Setting up complex wallets without that will be very slow

Reply:
ah, this is what we had discussed internally, right